src/adapter/vertex.rs

```diff
@@ -165,10 +173,17 @@ impl<'a> Vertex<'a> {
     }
 
     pub(super) fn as_variant(&self) -> Option<&'a Variant> {
-        self.as_item().and_then(|item| match &item.inner {
-            rustdoc_types::ItemEnum::Variant(v) => Some(v),
+        match &self.kind {
+            VertexKind::Variant(variant) => variant.variant(),
             _ => None,
-        })
+        }
+    }
```

Comment:
Consider refactoring the code so that we don't have two confusing methods that do almost-but-not-quite the same thing? We probably want `as_variant()` to return `Option<&'_ EnumVariant<'a>>` and then to refactor the code that needed the old signature.